<title>RasMol V2.6 Molecular Visualisation Program</title>
<h1>RasMol V2.6</h1><p>
<h2>Molecular Visualisation Program</h2><p>
<h3>Roger Sayle</h3>
<h3>Glaxo Wellcome Research and Development</h3>
<h3>Stevenage, Hertfordshire, U.K.</h3><p><p>
<h2>Table of Contents</h2>
<ol>
<li><a href="#chintro">Introduction</a>
<li><a href="#chcomref">Command Reference</a>
<li><a href="#chsetopt">Internal Parameters</a>
<li><a href="#chexprs">Atom Expressions</a>
<li><a href="#predefinedsets">Predefined Sets</a>
<li><a href="#chcolours">Colour Schemes</a>
</ol>
<p><hr>
&lt;p&gt;&lt;p&gt;Copyright (c) 1992-1995 by Roger Sayle
&lt;address&gt;rasmol@ggr.co.uk&lt;/address&gt;
<p>
The information supplied in this document is believed to be true but
no liability is assumed for its use or for the infringements of the
rights of the others resulting from its use.
<p>
Information in this document is subject to change without notice and does
not represent a commitment on the part of the supplier. This package is
sold/distributed subject to the condition that it shall not, by way of
trade or otherwise, be lent, re-sold, hired out or otherwise circulated
without the supplier's prior consent, in any form of packaging or cover
other than that in which it was produced. No part of this manual or
accompanying software may be reproduced, stored in a retrieval system on
optical or magnetic disk, tape or any other medium, or transmitted in any
form or by any means, electronic, mechanical, photocopying, recording or
otherwise for any purpose other than the purchaser's personal use.
<p>
This product is not to be used in the planning, construction, maintenance,
operation or use of any nuclear facility nor the flight, navigation or
communication of aircraft or ground support equipment. The author shall
not be liable, in whole or in part, for any claims or damages arising
from such use, including death, bancruptcy or outbreak of war.
<p><hr>
<a name="chintro"><h2>Introduction</h2></a>
RasMol2 is a molecular graphics program intended for the visualisation of
proteins, nucleic acids and small molecules. The program is aimed at
display, teaching and generation of publication quality images. RasMol
runs on Microsoft Windows, Apple Macintosh, UNIX and VMS systems. The
UNIX and VMS systems require an 8, 24 or 32 bit colour X Windows display
(X11R4 or later). The program reads in a molecule co-ordinate file and
interactively displays the molecule on the screen in a variety of colour
schemes and molecule representations. Currently available representations
include depth-cued wireframes, 'Dreiding' sticks, spacefilling (CPK) spheres,
ball and stick, solid and strand biomolecular ribbons, atom labels and dot
surfaces.
<p><hr>
<a name="chcomref"><h2>Command Reference</h2></a>
RasMol allows the execution of interactive commands typed at the
"<tt><b>RasMol></b></tt>"
prompt in the terminal window. Each command must be given on
a separate line. Keywords are case insensitive and may be entered in
either upper or lower case letters. All whitespace characters are
ignored except to separate keywords and their arguments.
<p>
The commands/keywords currently recognised by RasMol are given below.
<p>
<pre>
    <a href="#backbone">Backbone</a>        <a href="#background">Background</a>      <a href="#centre">Centre</a>
    <a href="#clipboard">Clipboard</a>       <a href="#colour">Colour</a>          <a href="#connect">Connect</a>
    <a href="#spacefill">CPK</a>             <a href="#dots">Dots</a>            <a href="#define">Define</a>
    <a href="#echo">Echo</a>            <a href="#quit">Exit</a>            <a href="#hbonds">HBonds</a>
    <a href="#help">Help</a>            <a href="#label">Label</a>           <a href="#load">Load</a>
    <a href="#print">Print</a>           <a href="#quit">Quit</a>            <a href="#renumber">Renumber</a>
    <a href="#reset">Reset</a>           <a href="#restrict">Restrict</a>        <a href="#ribbons">Ribbons</a>
    <a href="#rotate">Rotate</a>          <a href="#save">Save</a>            <a href="#script">Script</a>
    <a href="#select">Select</a>          <a href="#set">Set</a>             <a href="#show">Show</a>
    <a href="#slab">Slab</a>            <a href="#source">Source</a>          <a href="#spacefill">Spacefill</a>
    <a href="#ssbonds">SSBonds</a>         <a href="#strands">Strands</a>         <a href="#structure">Structure</a>
    <a href="#backbone">Trace</a>           <a href="#translate">Translate</a>       <a href="#wireframe">Wireframe</a>
    <a href="#write">Write</a>           <a href="#zap">Zap</a>             <a href="#zoom">Zoom</a>
</pre><hr>
<a name="backbone"><h3>Backbone</h3></a><p>
<pre>
Syntax:  backbone {&lt;boolean&gt;}
         backbone &lt;value&gt;
</pre><p>
The RasMol
<tt><b>backbone</b></tt>
command permits the representation of a polypeptide
backbone as a series of bonds connecting the adjacent alpha carbons of
each amino acid in a chain. The display of these backbone `bonds' is
turned on and off by the command paramater the same as the
<a href="#wireframe"><tt><b>wireframe</b></tt></a>
command. The command
<tt><b>backbone off</b></tt>
turns off the selected `bonds', and
<tt><b>backbone on</b></tt>
or with a number turns them on. The number can be used
to specify the cylinder radius of the representation in either angstrom
or rasmol units. A parameter value of 500 (2.0 angstroms) or above
results in a "Parameter value too large" error. Backbone objects may be
coloured using the RasMol
<a href="#colour"><tt><b>colour backbone</b></tt></a>
command.
<p>
The reserved work
<tt><b>backbone</b></tt>
is also used as a predefined set and as a parameter to the
<tt><b>set hbond</b></tt>
and
<tt><b>set ssbond</b></tt>
commands. The RasMol command
<tt><b>trace</b></tt>
is synonymous with the command
<tt><b>backbone.</b></tt>
<p><hr>
<a name="background"><h3>Background</h3></a><p>
<pre>
Syntax:  background &lt;colour&gt;
</pre><p>
The RasMol
<tt><b>background</b></tt>
command is used to set the colour of the "canvas" background. The
colour may be given as either a colour name or a comma separated
triple of Red, Green and Blue (RGB) components enclosed in square
brackets. Typing the command
<a href="#help"><tt><b>help colours</b></tt></a>
will give a list of the predefined colour names recognised by RasMol.
When running under X Windows, RasMol also recognises colours in the
X server's colour name database.
<p>
The
<tt><b>background</b></tt>
command is synonymous with the RasMol
<a href="#setbackground"><tt><b>set background</b></tt></a>
command.
<p><hr>
<a name="centre"><h3>Centre</h3></a><p>
<pre>
Syntax:  center {&lt;expression&gt;}
         centre {&lt;expression&gt;}
</pre><p>
The RasMol
<tt><b>centre</b></tt>
command defines the point about which the
<a href="#rotate"><tt><b>rotate</b></tt></a>
command and the scroll bars rotate the current molecule. Without a
parameter the centre command resets the centre of rotation to be the
centre of gravity of the molecule. If an atom expression is specified,
RasMol rotates the molecule about the centre of gravity of the set of
atoms specified by the expression. Hence, if a single atom is specified
by the expression, that atom will remain `stationary' during rotations.
<p>
Type
<a href="#help"><tt><b>help expression</b></tt></a>
for more information on RasMol atom expressions.
<p><hr>
<a name="clipboard"><h3>Clipboard</h3></a><p>
<pre>
Syntax:  clipboard
</pre><p>
The RasMol
<tt><b>clipboard</b></tt>
command places a copy of the currently displayed image on the local
graphics `clipboard'. Note: this command is not yet supported on
UNIX or VMS machines. It is intended to make transfering images
between applications easier under Microsoft Windows or on an Apple
Macintosh.
<p>
When using RasMol on a UNIX or VMS system this functionality may be
achieved by generating a raster image in a format that can be read
by the receiving program using the RasMol
<a href="#write"><tt><b>write</b></tt></a>
command.
<p><hr>
<a name="colour"><h3>Colour</h3></a><p>
<pre>
Syntax:  colour {&lt;object&gt;} &lt;colour&gt;
         color {&lt;object&gt;} &lt;colour&gt;
</pre><p>
Colour the atoms (or other objects) of the selected region. The colour may
be given as either a colour name or a comma separated triple of Red, Green
and Blue (RGB) components enclosed in square brackets. Typing the command
<a href="#help"><tt><b>help colours</b></tt></a>
will give a list of all the predefined colour names recognised
by RasMol.
<p>
Allowed objects are
<tt><b>atoms,</b></tt>
<tt><b>bonds,</b></tt>
<a href="#backbone"><tt><b>backbone,</b></tt></a>
<a href="#ribbons"><tt><b>ribbons</b></tt></a>
<a href="#labels"><tt><b>labels</b></tt></a>
<a href="#dots"><tt><b>dots,</b></tt></a>
<a href="#hbonds"><tt><b>hbonds,</b></tt></a>
and
<a href="#ssbonds"><tt><b>ssbonds.</b></tt></a>
If no object is specified, the default keyword
<tt><b>atom</b></tt>
is assumed.
Some colour schemes are defined for certain object types. The colour scheme
<tt><b>none</b></tt>
can be applied all objects accept atoms and dots, stating that the selected
objects have no colour of their own, but use the colour of their associated
atoms (i.e. the atoms they connect).
<tt><b>Atom</b></tt>
objects can also be coloured by
<a href="#cpkcolours"><tt><b>cpk,</b></tt></a>
<a href="#aminocolours"><tt><b>amino,</b></tt></a>
<a href="#chaincolours"><tt><b>chain,</b></tt></a>
<a href="#groupcolours"><tt><b>group,</b></tt></a>
<a href="#shapelycolours"><tt><b>shapely,</b></tt></a>
<a href="#structurecolours"><tt><b>structure,</b></tt></a>
<a href="#temperaturecolours"><tt><b>temperature</b></tt></a>
<a href="#chargecolours"><tt><b>charge</b></tt></a>
and
<a href="#usercolours"><tt><b>user. Hydrogen bonds can also be coloured by</b></tt></a>
<a href="#hbondtypecolours"><tt><b>type</b></tt></a>
and dot surfaces can also be coloured by
<a href="#potentialcolours"><tt><b>electrostatic potential.</b></tt></a>
For more information type
<a href="#chcolours"><tt><b>help colour <colour>.</b></tt></a>
<p><hr>
<a name="connect"><h3>Connect</h3></a><p>
<pre>
Syntax:  connect {&lt;boolean&gt;}
</pre><p>
The RasMol
<tt><b>connect</b></tt>
command is used to force RasMol to (re)calculate the connectivity
of the current molecule. If the original input file contained
connectivity information, this is discarded. The command
<tt><b>connect false</b></tt>
uses an extremely fast heuristic algorithmm that is suitable for
determing bonding in large bio-molecules such as proteins and
nucleic acids.  The command
<tt><b>connect true</b></tt>
uses a slower more accurate algorithm based upon covalent radii
that is more suitable for small molecules containing inorganic
elements or strained rings. If no parameters are given, RasMol
determines which algorithm to use based on the number of atoms
in the file. Greater than 255 atoms causes RasMol to use the
faster implementation. This is the method used to determine
bonding, if necessary, when a molecule is first read in using
the
<a href="#load"><tt><b>load</b></tt></a>
command.
<p><hr>
<a name="define"><h3>Define</h3></a><p>
<pre>
Syntax:  define &lt;identifier&gt; &lt;expression&gt;
</pre><p>
The RasMol
<tt><b>define</b></tt>
command allows the user to associate an arbitrary set of atoms with a
unique identifier. This allows the definition of user-defined sets. These
sets are declared statically, i.e. once defined the contents of the set
do not change, even if the expression defining them depends on the
current transformation and representation of the molecule.
<p><hr>
<a name="dots"><h3>Dots</h3></a><p>
<pre>
Syntax:  dots {&lt;boolean&gt;}
         dots &lt;value&gt;
</pre><p>
The RasMol
<tt><b>dots</b></tt>
command is used to generate a Van der Waal's dot surface around the
currently selected atoms. Dot surfaces display regularly spaced points
on a sphere of Van der Waals' radius about each selected atom. Dots that
would are `buried' within the Van der Waal's radius of any other atom
(selected or not) are not displayed.
The command
<tt><b>dots on</b></tt>
deletes any existing dot surface and generates a dots surface around
the currently selected atom set with a default dot density of 100. The
command
<tt><b>dots off</b></tt>
deletes any existing dot surface. The dot density may be
specified by providing a numeric parameter between 1 and 1000. This
value approximately corresponds to the number of dots on the surface
of a medium sized atom.
<p>
By default, the colour of each point on a dot surface is the colour
of it's closest atom at the time the surface is generated. The colour
of the whole dot surface may be changed using the
<a href="#colour"><tt><b>colour dots</b></tt></a>
command.
<p><hr>
<a name="echo"><h3>Echo</h3></a><p>
<pre>
Syntax:  echo {&lt;string&gt;}
</pre><p>
The RasMol
<tt><b>echo</b></tt>
command is used to display a message in the RasMol command/terminal
window. The string parameter may optionally be delimited in double
quote characters. If no parameter is specified, the
<tt><b>echo</b></tt>
command displays a blank line. This command is particularly useful
for displaying text from within a RasMol
<a href="#script"><tt><b>script</b></tt></a>
file.
<p><hr>
<a name="hbonds"><h3>HBonds</h3></a><p>
<pre>
Syntax:  hbonds {&lt;boolean&gt;}
         hbonds &lt;value&gt;
</pre><p>
The RasMol
<tt><b>hbond</b></tt>
command is used to represent the hydrogen bonding of the protein
molecule's backbone. This information is useful in assessing the
protein's secondary structure. Hydrogen bonds are represented as
either dotted lines or cylinders between the donor and acceptor
residues. The first time the
<tt><b>hbond</b></tt>
command is used, the program searches the structure of the
molecule to find hydrogen bonded residues and reports the number of bonds
to the user. The command
<tt><b>hbonds on</b></tt>
displays the selected `bonds' as dotted lines, and the
<tt><b>hbonds off</b></tt>
turns off their display. The colour of hbond objects may be changed
by the
<a href="#colour"><tt><b>colour hbond</b></tt></a>
command. Initially, each hydrogen bond has the colours of its connected
atoms.
<p>
By default the dotted lines are drawn between the accepting oxygen and
the donating nitrogen. By using the
<a href="#sethbonds"><tt><b>set hbonds</b></tt></a>
command the alpha carbon positions of the appropriate residues may be
used instead. This is especially useful when examining proteins in
backbone representation.
<p><hr>
<a name="help"><h3>Help</h3></a><p>
<pre>
Syntax:  help {&lt;topic&gt; {&lt;subtopic&gt;}}
         ? {&lt;topic&gt; {&lt;subtopic&gt;}}
</pre><p>
The RasMol
<tt><b>help</b></tt>
command provides on-line help on the given topic.
<p><hr>
<a name="label"><h3>Label</h3></a><p>
<pre>
Syntax:  label {&lt;string&gt;}
         label &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>label</b></tt>
command allows an arbitrary formatted text string to be
associated with each currently selected atom.  This string may contain
embedded `expansion specifiers' which display properties of the atom
being labelled. An expansion specifier consists of a `%' character
followed by a single alphabetic character specifying the property to be
displayed (similar to C's printf syntax).
An actual '%' character may be displayed by using the expansion
specifier `%%'.
<p>
Atom labelling for the currently selected atoms may be turned off with
the command
<tt><b>label off.</b></tt>
By default, if no string is given as a parameter RasMol uses labels
appropriate for the current molecule.
RasMol uses the label "%n%r:%c.%a" if the molecule contains more than
one chain, "%e%i" if the molecule has only a single residue (a small
molecule) and "%n%r.%a" otherwise.
<p>
The colour of each label may be changed using the
<a href="#colour"><tt><b>colour label</b></tt></a>
command. By default, each label is drawn in the same colour as the atom
to which it is attached. The size of the displayed text may be changed
using the
<a href="#setfontsize"><tt><b>set fontsize</b></tt></a>
command.
<p>
The following table lists the current expansion specifiers:
<p>
<pre>
    %a      Atom Name
    %b %t   B-factor/Temperature
    %c %s   Chain Identifier
    %e      Element Atomic Symbol
    %i      Atom Serial Number
    %n      Residue Name
    %r      Residue Number
</pre>
<p><hr>
<a name="load"><h3>Load</h3></a><p>
<pre>
Syntax:  load {&lt;format&gt;} &lt;filename&gt;
</pre><p>
Load a molecule co-ordinate file into RasMol2. Valid molecule file
formats are
<tt><b>pdb</b></tt>
(Brookhaven Protein Databank),
<tt><b>mdl</b></tt>
(Molecular Design Limited's MOL file format),
<tt><b>alchemy</b></tt>
(Tripos' Alchemy file format),
<tt><b>mol2</b></tt>
(Tripos' Sybyl Mol2 file format),
<tt><b>charmm</b></tt>
(CHARMm file format) or
<tt><b>xyz</b></tt>
(MSC's XMol XYZ file format). If no file format is specified,
<tt><b>pdb</b></tt>
is assumed by default. Only a single molecule may be loaded at a time.
To delete a molecule prior to loading another use the RasMol
<a href="#zap"><tt><b>zap</b></tt></a>
command.
<p>
The
<tt><b>load</b></tt>
command selects all the atoms in the molecule, centres it on the
screen and renders it as a CPK coloured wireframe model. If the molecule
contains no bonds (i.e. contains only alpha carbons), it is drawn as
an alpha carbon backbone. If the file specifies less bonds than atoms,
RasMol determines connectivity using the
<a href="#connect"><tt><b>connect</b></tt></a>
command.
<p><hr>
<a name="print"><h3>Print</h3></a><p>
<pre>
Syntax:  print
</pre><p>
The RasMol
<tt><b>print</b></tt>
command sends the currently displayed image to the local default printer
using the operating system's native printer driver. Note: this command
is not yet supported under UNIX or VMS. It is intended to take advantage
of Microsoft Windows and Apple Macintosh printer drivers. For example,
allowing images to be printed directly on a dot matrix printer.
<p>
When using RasMol on a UNIX or VMS system this functionality may be
achieved by either generating a PostScript file using the RasMol
<a href="#write"><tt><b>write ps</b></tt></a>
or
<a href="#write"><tt><b>write vectps</b></tt></a>
commands and printing that or generating a raster image file and using a
utility to dump that to the local printer.
<p><hr>
<a name="quit"><h3>Quit</h3></a><p>
<pre>
Syntax:  quit
         exit
</pre><p>
Exit from the RasMol program. The RasMol commands
<tt><b>exit</b></tt>
and
<tt><b>quit</b></tt>
are synonymous.
<p><hr>
<a name="renumber"><h3>Renumber</h3></a><p>
<pre>
Syntax:  renumber {{-} &lt;value&gt;}
</pre><p>
The RasMol
<tt><b>renumber</b></tt>
command sequentially numbers the residues in a macromolecular chain.
The optional parameter specifies the value of the first residue in the
sequence. By default, this value is one. For proteins,
each amino acid is numbered consecutively from the N terminus to the C
terminus. For nucleic acids, each base is numbered from the 5' terminus
to 3' terminus. All chains in the current database are renumbered and gaps
in the original sequence are ignored. The starting value for numbering may
be negative.
<p><hr>
<a name="reset"><h3>Reset</h3></a><p>
<pre>
Syntax:  reset
</pre><p>
The RasMol
<tt><b>reset</b></tt>
command restores the original viewing transformation
and centre of rotation. The scale is set to it default value,
<a href="#zoom"><tt><b>zoom 100,</b></tt></a>
the centre of rotation is set to the geometric centre of the currently
loaded molecule,
<a href="#centre"><tt><b>centre all,</b></tt></a>
this centre is translated to the middle of the screen and
the viewpoint set to the default orientation.
<p>
This command should not be mistaken for the RasMol
<a href="#zap"><tt><b>zap</b></tt></a>
command which deletes the currently stored molecule, returning the
program to its initial state.
<p><hr>
<a name="restrict"><h3>Restrict</h3></a><p>
<pre>
Syntax:  restrict {&lt;expression&gt;}
</pre><p>
The RasMol
<tt><b>restrict</b></tt>
command both defines the currently selected region of the
molecule and disables the representation of (most of) those parts of the
molecule no longer selected.  All subsequent RasMol commands that modify
a molecule's colour or representation effect only the currently selected
region. The parameter of a
<tt><b>restrict</b></tt>
command is a RasMol atom expression that is evaluated for every atom
of the current molecule. This command is very similar to the RasMol
<a href="#select"><tt><b>select</b></tt></a>
command, except restrict disables the
<a href="#wireframe"><tt><b>wireframe,</b></tt></a>
<a href="#spacefill"><tt><b>spacefill</b></tt></a>
and
<a href="#backbone"><tt><b>backbone</b></tt></a>
representations in the non-selected region.
<p>
Type "help expression" for more information on RasMol atom expressions.
<p><hr>
<a name="ribbons"><h3>Ribbons</h3></a><p>
<pre>
Syntax:  ribbons {&lt;boolean&gt;}
         ribbons &lt;value&gt;
</pre><p>
The RasMol
<tt><b>ribbons</b></tt>
command displays the currently loaded protein or nucleic acid as a
smooth solid "ribbon" surface passing along the backbone of the protein.
The ribbon is drawn between each amino acid whose alpha carbon is
currently selected. The colour of the ribbon is changed by the RasMol
<a href="#colour"><tt><b>colour ribbon</b></tt></a>
command. If the current ribbon colour is
<tt><b>none</b></tt>
(the default), the colour is taken from the alpha carbon at each
position along its length.
<p>
The width of the ribbon at each position is determined by the optional
parameter in the usual RasMol units. By default the width of the ribbon
is taken from the secondary structure of the protein or a constant value
of 720 (2.88 Angstroms) for nucleic acids.
The default width of protein alpha helices and beta sheets is 380 (1.52
Angstroms) and 100 (0.4 Angstroms) for turns and random coil. The
secondary structure assignment is either from the PDB file or calculated
using the DSSP algorithm as used by the
<a href="#structure"><tt><b>structure</b></tt></a>
command. This command is similar to the RasMol command
<a href="#strands"><tt><b>strands</b></tt></a>
which renders the biomolecular ribbon as parallel depth-cued curves.
<p><hr>
<a name="rotate"><h3>Rotate</h3></a><p>
<pre>
Syntax:  rotate &lt;axis&gt; {-} &lt;value&gt;
</pre><p>
Rotate the molecule about the specified axis.
Permited values for the axis parameter are
"&lt;tt&gt;&lt;b&gt;x&lt;/b&gt;&lt;/tt&gt;", "&lt;tt&gt;&lt;b&gt;y&lt;/b&gt;&lt;/tt&gt;" and "&lt;tt&gt;&lt;b&gt;z&lt;/b&gt;&lt;/tt&gt;".
The integer parameter states the angle in degrees for the structure to
be rotated. For the X and Y axes, positive values move the closest point
up and right, and negative values move it down and left respectively. For
the Z axis, a positive rotation acts clockwise and a negative angle
anti-clockwise.
<p><hr>
<a name="save"><h3>Save</h3></a><p>
<pre>
Syntax:  save {pdb} &lt;filename&gt;
         save alchemy &lt;filename&gt;
</pre><p>
Save the currently selected set of atoms in either a Brookhaven Protein
Database (PDB) or Alchemy(tm) format file.
The distinction between this command and the RasMol
<a href="#write"><tt><b>write</b></tt></a>
command has been dropped. The only difference is that without a format
specifier the
<tt><b>save</b></tt>
command generates a
<tt><b>PDB</b></tt>
file and the
<a href="#write"><tt><b>write</b></tt></a>
command generates a
<tt><b>GIF</b></tt>
image.
<p><hr>
<a name="script"><h3>Script</h3></a><p>
<pre>
Syntax:  script &lt;filename&gt;
</pre><p>
The RasMol
<tt><b>script</b></tt>
command reads a set of RasMol commands sequentially from a
text file and executes them. This allows sequences of commonly used
commands to be stored and performed by single command. A RasMol script
file may contain a further script command up to a maximum "depth" of 10,
allowing compilicated sequences of actions to be executed. RasMol
ignores all characters after the first '#' character on each line
allowing the scripts to be annotated. Script files are often also
annotated using the RasMol
<a href="#echo"><tt><b>echo</b></tt></a>
command.
<p>
The most common way to generate a RasMol script file is to use the
<a href="#write"><tt><b>write script</b></tt></a>
or
<a href="#write"><tt><b>write rasmol</b></tt></a>
commands to output the sequence of commands that are needed to
regenerate the current view, representation and colouring of the
currently displayed molecule.
<p>
The RasMol command
<tt><b>source</b></tt>
is synonymous with the
<tt><b>script</b></tt>
command.
<p><hr>
<a name="select"><h3>Select</h3></a><p>
<pre>
Syntax:  select {&lt;expression&gt;}
</pre><p>
Define the currently selected region of the molecule. All subsequent RasMol
commands that manipulate a molecule or modify its colour or representation,
only effects the currently selected region. The parameter of a
<tt><b>select</b></tt>
command is a RasMol expression that is evaluated for every atom of the
current molecule. The currently selected (active) region of the molecule
are those atoms that cause the expression to evaluate true. To select
the whole molecule use the RasMol command
<tt><b>select all.</b></tt>
The behaviour of the
<tt><b>select</b></tt>
command without any parameters is determined by the RasMol
<a href="#sethetero"><tt><b>hetero</b></tt></a>
and
<a href="#set"><tt><b>hydrogen</b></tt></a>
parameters.
<p>
Type "help expression" for more information on RasMol atom expressions.
<p><hr>
<a name="set"><h3>Set</h3></a><p>
<pre>
Syntax:  set &lt;parameter&gt; {&lt;option&gt;}
</pre><p>
The RasMol
<tt><b>set</b></tt>
command allows the user to alter various internal program parameters
such as those controlling rendering options. Each parameter has its
own set or permissible parameter options. Typically, ommiting the
paramter option resets that parameter to its default value. A list of
valid parameter names is given below.
<p>
<pre>
    <a href="#setambient">Ambient</a>         <a href="#setaxes">Axes</a>            <a href="#setbackground">Background</a>
    <a href="#setbondmode">BondMode</a>        <a href="#setboundbox">BoundBox</a>        <a href="#setdisplay">Display</a>
    <a href="#setfontsize">FontSize</a>        <a href="#sethbonds">HBonds</a>          <a href="#sethetero">Hetero</a>
    <a href="#sethourglass">HourGlass</a>       <a href="#sethydrogen">Hydrogen</a>        <a href="#setkinemage">Kinemage</a>
    <a href="#setmenus">Menus</a>           <a href="#setmouse">Mouse</a>           <a href="#setradius">Radius</a>
    <a href="#setshadow">Shadow</a>          <a href="#setslabmode">SlabMode</a>        <a href="#setsolvent">Solvent</a>
    <a href="#setspecular">Specular</a>        <a href="#setspecpower">SpecPower</a>       <a href="#setssbonds">SSBonds</a>
    <a href="#setstrands">Strands</a>         <a href="#setunitcell">UnitCell</a>        <a href="#setvectps">VectPS</a>
</pre>
<p><hr>
<a name="show"><h3>Show</h3></a><p>
<pre>
Syntax:  show information
         show sequence
         show symmetry
</pre><p>
The RasMol
<tt><b>show</b></tt>
command display details of the status of the currently
loaded molecule. The command
<tt><b>show information</b></tt>
lists the molecule's name,
classification, PDB code and the number of atoms, chains, groups it contains.
If hydrogen bonding, disulphide bridges or secondary structure have been
determined, the number of hbonds, ssbonds, helices, ladders and turns
are also displayed respectively. The command
<tt><b>show sequence</b></tt>
lists the residues that compose each chain of the molecule.
<p><hr>
<a name="slab"><h3>Slab</h3></a><p>
<pre>
Syntax:  slab {&lt;boolean&gt;}
         slab &lt;value&gt;
</pre><p>
The RasMol
<tt><b>slab</b></tt>
command enables, disables or positions the z-clipping plane of the
molecule. The program only draws those portions of the
molecule that are further from the viewer than the slabbing plane.
Values range  from zero at the very back of the molecule to
100 which is completely in front of the molecule. Intermediate values
determine the percentage of the molecule to be drawn.
<p><hr>
<a name="spacefill"><h3>Spacefill</h3></a><p>
<pre>
Syntax:  spacefill {&lt;boolean&gt;}
         spacefill temperature
         spacefill user
         spacefill &lt;value&gt;
</pre><p>
The RasMol
<tt><b>spacefill</b></tt>
command is used to represent all of the currently selected atoms as solid
spheres. This command is used to produce both union-of-spheres and
ball-and-stick models of a molecule. The command,
<tt><b>spacefilll true,</b></tt>
the default, represents each atom as a sphere of Van der Waals radius.
The command
<tt><b>spacefill off</b></tt>
turns off the representation of the selected atom as spheres. A sphere
radius may be specified as an integer in RasMol units (1/250th Angstrom)
or a value containing a decimal point. A value of 500 (2.0
Angstroms) or greater results in a "Parameter value too large" error.
<p>
The
<tt><b>temperature</b></tt>
option sets the radius of each sphere to the value stored in its temperature
field. Zero or negative values causes have no effect and values greater than
2.0 are truncated to 2. The
<tt><b>user</b></tt>
option allows the radius of each spheres to be specified by additional lines
in the molecule's PDB file using Raster 3D's COLOR record extension.
<p>
The RasMol command
<tt><b>cpk</b></tt>
is synonymous with the
<tt><b>spacefill</b></tt>
command.
<p><hr>
<a name="ssbonds"><h3>SSBonds</h3></a><p>
<pre>
Syntax:  ssbonds {&lt;boolean&gt;}
         ssbonds &lt;value&gt;
</pre><p>
The RasMol
<tt><b>ssbonds</b></tt>
command is used to represent the disulphide bridges of the protein
molecule as either dotted lines or cylinders between the connected
cysteines. The first time that the
<tt><b>ssbonds</b></tt>
command is used, the program searches the structure of the protein to
find half-cysteine pairs (cysteines whose sulphurs are within 3 angstroms
of each other) and reports the number of bridges to the user. The command
<tt><b>ssbonds on</b></tt>
displays the selected `bonds' as dotted lines, and the command
<tt><b>ssbonds off</b></tt>
disables the display of ssbonds in the currently selected area. Selection
of disulphide bridges is identical to normal bonds, and may be adjusted
using the RasMol
<a href="#setbondmode"><tt><b>set bondmode</b></tt></a>
command. The colour of disulphide bonds may be changed using the
<a href="#colour"><tt><b>colour ssbonds</b></tt></a>
command. By default, each disulphide bond has the colours of its connected
atoms.
<p>
By default disulphide bonds are drawn between the sulphur atoms within
the cysteine groups. By using the
<a href="#setssbonds"><tt><b>set ssbonds</b></tt></a>
command the position of the cysteine's alpha carbons may be used instead.
<p><hr>
<a name="strands"><h3>Strands</h3></a><p>
<pre>
Syntax:  strands {&lt;boolean&gt;}
         strands &lt;value&gt;
</pre><p>
The RasMol
<tt><b>strands</b></tt>
command displays the currently loaded protein or nucleic acid as a
smooth "ribbon" of depth-cued curves passing along the backbone of the
protein. The ribbon is composed of a number of strands that run parallel
to one another along the peptide plane of each residue. The ribbon is
drawn between each amino acid whose alpha carbon is currently selected.
The colour of the ribbon is changed by the RasMol
<a href="#colour"><tt><b>colour ribbon</b></tt></a>
command. If the current ribbon colour is
<tt><b>none</b></tt>
(the default), the colour is taken from the alpha carbon at each
position along its length. The colour of the central and outermost
strands may be coloured independently using the
<a href="#colour"><tt><b>colour ribbon1</b></tt></a>
and
<a href="#colour"><tt><b>colour ribbon2</b></tt></a>
commands respectively. The number of strands in the ribbon may be
altered using the RasMol
<a href="#setstrands"><tt><b>set strands</b></tt></a>
command.
<p>
The width of the ribbon at each position is determined by the optional
parameter in the usual RasMol units. By default the width of the ribbon
is taken from the secondary structure of the protein or a constant value
of 720 for nucleic acids (which produces a ribbon 2.88 Angstroms wide).
The default width of protein alpha helices and beta sheets is 380 (1.52
Angstroms) and 100 (0.4 Angstroms) for turns and random coil. The
secondary structure assignment is either from the PDB file or calculated
using the DSSP algorithm as used by the
<a href="#structure"><tt><b>structure</b></tt></a>
command. This command is similar to the RasMol command
<a href="#ribbons"><tt><b>ribbons</b></tt></a>
which renders the biomolecular ribbon as a smooth shaded surface.
<p><hr>
<a name="structure"><h3>Structure</h3></a><p>
<pre>
Syntax:  structure
</pre><p>
The RasMol
<tt><b>structure</b></tt>
command calculates secondary structure assignments
for the currently loaded protein. If the original PDB file contained
structural assignment records (HELIX and SHEET) these are discarded.
Initially, the hydrogen bonds of the current molecule are found, if this
hasn't been done already. The secondary structure is the determined using
Kabsch and Sander's DSSP algorithm. Once finished the program reports the
number of helices, strands and turns found.
<p><hr>
<a name="translate"><h3>Translate</h3></a><p>
<pre>
Syntax:  translate &lt;axis&gt; {-} &lt;value&gt;
</pre><p>
The RasMol
<tt><b>translate</b></tt>
command moves the position of the centre of the molecule on the
screen. The axis parameter specifies along which axis the molecule
is to be moved and the integer parameter specifies the absolute
position of the molecule centre from the middle of the screen.
Permited values for the axis parameter are
"&lt;tt&gt;&lt;b&gt;x&lt;/b&gt;&lt;/tt&gt;", "&lt;tt&gt;&lt;b&gt;y&lt;/b&gt;&lt;/tt&gt;" and "&lt;tt&gt;&lt;b&gt;z&lt;/b&gt;&lt;/tt&gt;".
Displacement values must be between -100 and 100 which correspond to
moving the current molecule just off the screen. A positive
"&lt;tt&gt;&lt;b&gt;x&lt;/b&gt;&lt;/tt&gt;"
displacement moves the molecule to the right, and a positive
"&lt;tt&gt;&lt;b&gt;y&lt;/b&gt;&lt;/tt&gt;"
displacement moves the molecule down the screen. The pair of commands
<tt><b>translate x 0</b></tt>
and
<tt><b>translate y 0</b></tt>
centres the molecule on the screen.
<p><hr>
<a name="wireframe"><h3>Wireframe</h3></a><p>
<pre>
Syntax:  wireframe {&lt;boolean&gt;}
         wireframe &lt;value&gt;
</pre><p>
The RasMol
<tt><b>wireframe</b></tt>
command represents each bond within the selected region of the molecule
as either a cylinder, a line or depth-cued vector. The display of bonds
as depth-cued vectors (drawn darker the further away from the viewer)
is turned on by the command
<tt><b>wireframe</b></tt>
or
<tt><b>wireframe on.</b></tt>
The selected bonds are displayed as cylinders by specifying a radius
either as an integer in RasMol units or containing a decimal point as
a value in Angstroms.  A parameter value of 500 (2.0 angstroms) or
above results in an "Parameter value too large" error. Bonds may be
coloured using the
<a href="#colour"><tt><b>colour bonds</b></tt></a>
command.
<p><hr>
<a name="write"><h3>Write</h3></a><p>
<pre>
Syntax:  write {&lt;format&gt;} &lt;filename&gt;
</pre><p>
Write the current image to a file in a standard raster format. Currently
supported image file formats include
"<tt><b>gif</b></tt>"
(Compuserve GIF),
"<tt><b>ppm</b></tt>"
(Portable Pixmap),
"<tt><b>ras</b></tt>"
(Sun rasterfile),
"<tt><b>ps</b></tt>"
and
"<tt><b>epsf</b></tt>"
(Encapsulated PostScript),
"<tt><b>monops</b></tt>"
(Monochrome Encapsulated PostScript),
"<tt><b>bmp</b></tt>"
(Microsoft bitmap) and
"<tt><b>pict</b></tt>"
(Apple PICT). The
<tt><b>write</b></tt>
command may also be used to generate command scripts for other graphics
programs. The format
<tt><b>script</b></tt>
writes out a file containing the RasMol
<a href="#script"><tt><b>script</b></tt></a>
commands to reproduce the current image. The format
<tt><b>molscript</b></tt>
writes out the commands required to render the current view of the
molecule as ribbons in Per Kraulis' Molscript program and the format
<tt><b>kinemage</b></tt>
the commands for David Richardson's program Mage.
<p>
The distinction between this command and the RasMol
<a href="#save"><tt><b>save</b></tt></a>
command has been dropped. The only difference is that without a format
specifier the
<a href="#save"><tt><b>save</b></tt></a>
command generates a
<tt><b>PDB</b></tt>
file and the
<tt><b>write</b></tt>
command generates a
<tt><b>GIF</b></tt>
image.
<p><hr>
<a name="zap"><h3>Zap</h3></a><p>
<pre>
Syntax:  zap
</pre><p>
Deletes the contents of the current database and resets parameter
variables to their initial default state.
<p><hr>
<a name="zoom"><h3>Zoom</h3></a><p>
<pre>
Syntax:  zoom {&lt;boolean&gt;}
         zoom &lt;value&gt;
</pre><p>
Change the magnification of the currently displayed image. Boolean
parameters either magnify or reset the scale of current molecule. An
integer parameter specifies the desired magnification as a percentage
of the default scale. The minimum parameter value is 10, the maximum
parameter value is dependent upon the size of the molecule being
displayed. For medium sized proteins this is about 500.
<p><hr>
<a name="chsetopt"><h2>Internal Parameters</h2></a>
RasMol has a number of internal parameters that may be modified using the
<a href="#set"><tt><b>set</b></tt></a>
command. These parameters control a number of program options such as
rendering options and mouse button mappings.
<p>
A complete list of internal parameter names is given below.
<p>
<pre>
    <a href="#setambient">Ambient</a>         <a href="#setaxes">Axes</a>            <a href="#setbackground">Background</a>
    <a href="#setbondmode">BondMode</a>        <a href="#setboundbox">BoundBox</a>        <a href="#setdisplay">Display</a>
    <a href="#setfontsize">FontSize</a>        <a href="#sethbonds">HBonds</a>          <a href="#sethetero">Hetero</a>
    <a href="#sethourglass">HourGlass</a>       <a href="#sethydrogen">Hydrogen</a>        <a href="#setkinemage">Kinemage</a>
    <a href="#setmenus">Menus</a>           <a href="#setmouse">Mouse</a>           <a href="#setradius">Radius</a>
    <a href="#setshadow">Shadow</a>          <a href="#setslabmode">SlabMode</a>        <a href="#setsolvent">Solvent</a>
    <a href="#setspecular">Specular</a>        <a href="#setspecpower">SpecPower</a>       <a href="#setssbonds">SSBonds</a>
    <a href="#setstrands">Strands</a>         <a href="#setunitcell">UnitCell</a>        <a href="#setvectps">VectPS</a>
</pre>
<p><hr>
<a name="setambient"><h3>Set Ambient</h3></a><p>
<pre>
Syntax:  set ambient {&lt;value&gt;}
</pre><p>
The RasMol
<tt><b>ambient</b></tt>
parameter is used to control the amount of ambient (or surrounding)
light in the scene. The
<tt><b>ambient</b></tt>
value must be between 0 and 100 that controls the percentage intensity
of the darkest shade of an object. For a solid object, this is the
intensity of surfaces facing away from the light source or in shadow.
For depth-cued objects this is the intensity of objects furthest from
the viewer.
<p>
This parameter is commonly used to correct for monitors with different
"gamma values" (brightness), to change how light or dark a hardcopy
image appears when printed or to alter the feeling of depth for
wireframe or ribbon representations.
<p><hr>
<a name="setaxes"><h3>Set Axes</h3></a><p>
<pre>
Syntax:  set axes &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>axes</b></tt>
parameter controls the display of orthogonal co-ordinate axes on
the current display. The co-ordinate axes are those used in the
molecule data file, and the origin is the centre of the molecule's
bounding box. The
<tt><b>set axes</b></tt>
command is similar the the commands
<a href="#setboundbox"><tt><b>set boundbox</b></tt></a>
and
<a href="#setunitcell"><tt><b>set unitcell</b></tt></a>
that display the bounding box and the crystallographic unit cell
respectively.
<p><hr>
<a name="setbackground"><h3>Set Background</h3></a><p>
<pre>
Syntax:  set background {&lt;colour&gt;}
</pre><p>
The RasMol
<tt><b>background</b></tt>
parameter is used to set the colour of the "canvas" background. The
colour may be given as either a colour name or a comma separated
triple of Red, Green, Blue (RGB) components enclosed in square
brackets. Typing the command
<a href="#help"><tt><b>help colours</b></tt></a>
will give a list of the predefined colour names recognised by RasMol.
When running under X Windows, RasMol also recognises colours in the
X server's colour name database.
<p>
The command
<tt><b>set background</b></tt>
is synonymous with the RasMol command
<a href="#background"><tt><b>background.</b></tt></a>
<p><hr>
<a name="setbondmode"><h3>Set BondMode</h3></a><p>
<pre>
Syntax:  set bondmode and
         set bondmode or
</pre><p>
The RasMol
<tt><b>set bondmode</b></tt>
command controls the mechanism used to select individual bonds. When
using the
<a href="#select"><tt><b>select</b></tt></a>
and
<a href="#restrict"><tt><b>restrict</b></tt></a>
commands, a given bond will be selected if i) the bondmode is
<tt><b>or</b></tt>
and either of the connected atoms is selected, or ii) the bondmode is
<tt><b>and</b></tt>
and both atoms connected by the bond are selected. Hence an individual
bond may be uniquely identified by using the command
"<tt><b>set bondmode and</b></tt>"
and then uniquely selecting the atoms at both ends.
<p><hr>
<a name="setboundbox"><h3>Set BoundBox</h3></a><p>
<pre>
Syntax:  set boundbox &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>boundbox</b></tt>
parameter controls the display of the current molecules bounding box
on the display. The bounding box is orthogonal to the data file's
original co-ordinate axes. The
<tt><b>set boundbox</b></tt>
command is similar the the commands
<a href="#setaxes"><tt><b>set axes</b></tt></a>
and
<a href="#setboundbox"><tt><b>set unitcell</b></tt></a>
that display orthogonal co-ordinate axes and the bounding box
respectively.
<p><hr>
<a name="setdisplay"><h3>Set Display</h3></a><p>
<pre>
Syntax:  set display selected
         set display normal
</pre><p>
This command controls the display mode within RasMol. By default,
<tt><b>set display normal,</b></tt>
RasMol displays the molecule in the representation specified by the
user. The command
<tt><b>set display selected</b></tt>
changes the display mode such that the molecule is temporarily drawn
so as to indicate currently selected portion of the molecule. The
user specified colour scheme and representation remains unchanged.
In this representation all selected atoms are shown in yellow and
all non selected atoms are shown in blue. The colour of the background
is also changed to a dark grey to indicate the change of display mode.
This command is typically only used by external Graphical User
Interfaces (GUIs).
<p><hr>
<a name="sethbonds"><h3>Set HBonds</h3></a><p>
<pre>
Syntax:  set hbonds backbone
         set hbonds sidechain
</pre><p>
The RasMol
<tt><b>hbonds</b></tt>
parameter determines whether hydrogen bonds are drawn between
the donor and acceptor atoms of the hydrogen bond,
<tt><b>set hbonds sidechain</b></tt>
or between the alpha carbon atoms of the protein backbone and between
the phosphorous atoms of the nucleic acid backbone,
<tt><b>set hbonds backbone.</b></tt>
The actual display of hydrogen bonds is controlled by the
<a href="#hbonds"><tt><b>hbonds</b></tt></a>
command. Drawing hydrogen bonds between protein alpha carbons or
nucleic acid phosphorous atoms is useful when the rest of the molecule
is shown in only a schematic representation such as
<a href="#backbone"><tt><b>backbone,</b></tt></a>
<a href="#ribbons"><tt><b>ribbons</b></tt></a>
or
<a href="#strands"><tt><b>strands.</b></tt></a>
his parameter is similar to the RasMol
<a href="#setssbonds"><tt><b>ssbonds</b></tt></a>
parameter.
<p><hr>
<a name="setfontsize"><h3>Set FontSize</h3></a><p>
<pre>
Syntax:  set fontsize {&lt;value&gt;}
</pre><p>
The RasMol
<tt><b>set fontsize</b></tt>
command is used to control the size of the characters that
form atom labels. This value corresponds to the height of
the displayed character in pixels. The maximum value of
<tt><b>fontsize</b></tt>
is 32 pixels, and the default value is 8 pixels high. To
display atom labels on the screen use the RasMol
<a href="#label"><tt><b>label</b></tt></a>
command and to change the colour of displayed labels, use
the
<a href="#colour"><tt><b>colour labels</b></tt></a>
command.
<p><hr>
<a name="sethetero"><h3>Set Hetero</h3></a><p>
<pre>
Syntax:  set hetero &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>hetero</b></tt>
parameter is used to modify the `default' behaviour of the RasMol
<a href="#select"><tt><b>select</b></tt></a>
command, i.e. the behaviour of
<tt><b>select</b></tt>
without any parameters. When this value is
<tt><b>false,</b></tt>
the default
<a href="#select"><tt><b>select</b></tt></a>
region does not include an heterogenous atoms (refer to the
predefined set
<a href="#heteroset"><tt><b>hetero</b></tt></a>
). When this value is
<tt><b>true,</b></tt>
the default
<a href="#select"><tt><b>select</b></tt></a>
region may contain hetero atoms. This parameter is similar to
the RasMol
<a href="#sethydrogen"><tt><b>hydrogen</b></tt></a>
parameter which determines whether hydrogen atoms should be
included in the default set. If both
<tt><b>hetero</b></tt>
and
<a href="#sethydrogen"><tt><b>hydrogen</b></tt></a>
are
<tt><b>true,</b></tt>
<a href="#select"><tt><b>select</b></tt></a>
without any parameters is equivalent to
<a href="#select"><tt><b>select all.</b></tt></a>
<p><hr>
<a name="sethourglass"><h3>Set HourGlass</h3></a><p>
<pre>
Syntax:  set hourglass &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>hourglass</b></tt>
parameter allows the user to enable and disable the use of the `hour
glass' cursor used by RasMol to indicate that the program is currently
busy drawing the next frame. The command
<tt><b>set hourglass on</b></tt>
enable the indicator, whilst
<tt><b>set hourglass off</b></tt>
prevents RasMol from changing the cursor. This is useful when spinning
the molecule, running a sequence of commands from a script file or
using interprocess communication to execute complex sequences of
commands. In these cases a `flashing' cursor may be distracting.
<p><hr>
<a name="sethydrogen"><h3>Set Hydrogen</h3></a><p>
<pre>
Syntax:  set hydrogen &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>hydrogen</b></tt>
parameter is used to modify the `default' behaviour of the RasMol
<a href="#select"><tt><b>select</b></tt></a>
command, i.e. the behaviour of
<tt><b>select</b></tt>
without any parameters. When this value is
<tt><b>false,</b></tt>
the default
<a href="#select"><tt><b>select</b></tt></a>
region does not include any hydrogen or deuterium atoms (refer
to the predefined set
<a href="#hydrogenset"><tt><b>hydrogen</b></tt></a>
). When this value is
<tt><b>true,</b></tt>
the default
<a href="#select"><tt><b>select</b></tt></a>
region may contain hydrogen atoms. This parameter is similar to
the RasMol
<a href="#sethetero"><tt><b>hetero</b></tt></a>
parameter which determines whether heterogenous atoms should be
included in the default set. If both
<tt><b>hydrogen</b></tt>
and
<a href="#sethetero"><tt><b>hetero</b></tt></a>
are
<tt><b>true,</b></tt>
<a href="#select"><tt><b>select</b></tt></a>
without any parameters is equivalent to
<a href="#select"><tt><b>select all.</b></tt></a>
<p><hr>
<a name="setkinemage"><h3>Set Kinemage</h3></a><p>
<pre>
Syntax:  set kinemage &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>set kinemage</b></tt>
command controls the amount of detail stored in a Kinemage output
file generated by the RasMol
<a href="#write"><tt><b>write kinemage</b></tt></a>
command. The output kinemage files are intended to be displayed by
David Richardson's Mage program.
<tt><b>set kinemage false,</b></tt>
the default, only stores the currently displayed representation in
the generated output file. The command
<tt><b>set kinemage true,</b></tt>
generates a more complex Kinemage that contains both the wireframe
and backbone representations as well as the co-ordinate axes,
bounding box and crystal unit cell.
<p><hr>
<a name="setmenus"><h3>Set Menus</h3></a><p>
<pre>
Syntax:  set menus &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>set menus</b></tt>
command enables the canvas window's menu buttons or menu bar. This
command is typically only used by graphical user interfaces or to
create as large as image as possible when using Microsoft Windows.
<p><hr>
<a name="setmouse"><h3>Set Mouse</h3></a><p>
<pre>
Syntax:  set mouse rasmol
         set mouse insight
         set mouse quanta
</pre><p>
The RasMol
<tt><b>set mouse</b></tt>
command sets the rotation, translation, scaling and zooming mouse
bindings. The default value is
<tt><b>rasmol</b></tt>
which is suitable for two button mice (for three button mice the
second and third buttons are synonymous); X-Y rotation is controlled
by the first button, and X-Y translation by the second. Additional
functions are controlled by holding a modifier key on the keyboard.
[Shift] and the first button performs scaling, [shift] and the second
button performs Z-rotation, and [control] and the first mouse button
controls the clipping plane. The
<tt><b>insight</b></tt>
and
<tt><b>quanta</b></tt>
provide the same mouse bindings as other packages for experienced
users.
<p><hr>
<a name="setradius"><h3>Set Radius</h3></a><p>
<pre>
Syntax:  set radius {&lt;value&gt;}
</pre><p>
The RasMol
<tt><b>set radius</b></tt>
command is used to alter the behaviour of the RasMol
<a href="#dots"><tt><b>dots</b></tt></a>
command depending upon the value of the
<a href="#setsolvent"><tt><b>solvent</b></tt></a>
parameter.
When
<a href="#setsolvent"><tt><b>solvent</b></tt></a>
is
<tt><b>true,</b></tt>
the
<tt><b>radius</b></tt>
parameter controls whether a true Van der Waal's surface
is generated by the
<a href="#dots"><tt><b>dots</b></tt></a>
command. If the value of
<tt><b>radius</b></tt>
is anything other than zero, that value is used as the
radius of each atom instead of it true VdW value. When
the value of
<a href="#setsolvent"><tt><b>solvent</b></tt></a>
is
<tt><b>true,</b></tt>
this parameter determines the `probe sphere' (solvent) radius.
The parameter may be given as an integer in rasmol units or
containing a decimal point in Angstroms. The default value of
this parameter is determined by the value of
<a href="#setsolvent"><tt><b>solvent</b></tt></a>
and changing
<a href="#setsolvent"><tt><b>solvent</b></tt></a>
resets
<tt><b>radius</b></tt>
to its new default value.
<p><hr>
<a name="setshadow"><h3>Set Shadow</h3></a><p>
<pre>
Syntax:  set shadow &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>set shadow</b></tt>
command enables and disables raytracing of the currently rendered image.
Currently only the spacefilling representation is shadowed or can cast
shadows. Enabling shadowing will automatically disable the Z-clipping
(slabbing) plane using the command
<a href="#slab"><tt><b>slab off.</b></tt></a>
Raytracing typically takes about 10s for a moderately sized protein.
It is recommended that shadowing is normally disabled whilst the
molecule is being transformed or manipulated, and only enabled once
an appropiate viewpoint is selected, to provide a greater impression
of depth.
<p><hr>
<a name="setslabmode"><h3>Set SlabMode</h3></a><p>
<pre>
Syntax:  set slabmode &lt;slabmode&gt;
</pre><p>
The RasMol
<tt><b>slabmode</b></tt>
parameter controls the rendering method of objects cut by the
slabbing (z-clipping) plane. Valid slabmode parameters are
"&lt;tt&gt;&lt;b&gt;reject&lt;/b&gt;&lt;/tt&gt;", "&lt;tt&gt;&lt;b&gt;half&lt;/b&gt;&lt;/tt&gt;",
"&lt;tt&gt;&lt;b&gt;hollow&lt;/b&gt;&lt;/tt&gt;", "&lt;tt&gt;&lt;b&gt;solid&lt;/b&gt;&lt;/tt&gt;" and
"&lt;tt&gt;&lt;b&gt;section&lt;/b&gt;&lt;/tt&gt;".
<p><hr>
<a name="setsolvent"><h3>Set Solvent</h3></a><p>
<pre>
Syntax:  set solvent &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>set solvent</b></tt>
command is used to control the behaviour of the RasMol
<a href="#dots"><tt><b>dots</b></tt></a>
command. Depending upon the value of the
<tt><b>solvent</b></tt>
parameter, the
<a href="#dots"><tt><b>dots</b></tt></a>
command either generates a Van der Waal's or a solvent
acessible surface around the currently selected set of
atoms. Changing this parameter automatically resets the
value of the RasMol
<a href="#setradius"><tt><b>radius</b></tt></a>
parameter.
The command
<tt><b>set solvent false,</b></tt>
the default value, indicates that a Van der Waal's surface
should be generated and resets the value of
<a href="#setradius"><tt><b>radius</b></tt></a>
to zero. The command
<tt><b>set solvent true</b></tt>
indicates that a `Connolly' or `Richards' solvent
accessible surface should be drawn and sets the
<a href="#setradius"><tt><b>radius</b></tt></a>
parameter, the solvent radius, to 1.2 Angstroms (or 300
RasMol units).
<p><hr>
<a name="setspecular"><h3>Set Specular</h3></a><p>
<pre>
Syntax:  set specular &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>set specular</b></tt>
command enables and disables the display of specular highlights on
solid objects drawn by RasMol. Specular highlights appear as white
reflections of the light source on the surface of the object. The
current RasMol implementation uses an approximation function to
generate this highlight.
<p>
The specular highlights on the surfaces of solid objects may be
altered by using the specular reflection coefficient, which is
altered using the RasMol
<a href="#setspecpower"><tt><b>set specpower</b></tt></a>
command.
<p><hr>
<a name="setspecpower"><h3>Set SpecPower</h3></a><p>
<pre>
Syntax:  set specpower {&lt;value&gt;}
</pre><p>
The
<tt><b>specpower</b></tt>
parameter determines the shininess of solid objects rendered by
RasMol. This value between 0 and 100 adjusts the reflection
coeffient used in specular highlight calculations. The specular
highlights are enabled and disabled by the RasMol
<a href="#setspecular"><tt><b>set specular</b></tt></a>
command. Values around 20 or 30 produce plastic looking surfaces.
High values represent more shiny surfaces such as metals, while
lower values produce more diffuse/dull surfaces.
<p><hr>
<a name="setssbonds"><h3>Set SSBonds</h3></a><p>
<pre>
Syntax:  set ssbonds backbone
         set ssbonds sidechain
</pre><p>
The RasMol
<tt><b>ssbonds</b></tt>
parameter determines whether disulphide bridges are drawn between
the sulphur atoms in the sidechain (the default) or between the alpha
carbon atoms in the backbone of the cysteines residues. The actual
display of disulphide bridges is controlled by the
<a href="#ssbonds"><tt><b>ssbonds</b></tt></a>
command. Drawing disulphide bridges between alpha carbons is useful
when the rest of the protein is shown in only a schematic
representation such as
<a href="#backbone"><tt><b>backbone,</b></tt></a>
<a href="#ribbons"><tt><b>ribbons</b></tt></a>
or
<a href="#strands"><tt><b>strands.</b></tt></a>
his parameter is similar to the RasMol
<a href="#sethbonds"><tt><b>hbonds</b></tt></a>
parameter.
<p><hr>
<a name="setstrands"><h3>Set Strands</h3></a><p>
<pre>
Syntax:  set strands {&lt;value&gt;}
</pre><p>
The RasMol
<tt><b>strands</b></tt>
parameter controls the number of parallel strands that are displayed
in the ribbon representations of proteins. The permissible values for
this parameter are 1, 2, 3, 4, 5 and 9. The default value is 5. The
number of strands is constant for all ribbons being displayed.
However, the ribbon width (the separation between strands) may be
controlled on a residue by residue basis using the RasMol
<a href="#ribbons"><tt><b>ribbons</b></tt></a>
command.
<p><hr>
<a name="setunitcell"><h3>Set UnitCell</h3></a><p>
<pre>
Syntax:  set unitcell &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>unitcell</b></tt>
parameter controls the display of the crystallographic unit cell on
the current display. The crystal cell is only enabled if the appropriate
crystal symmetry information is contained in the PDB data file. The
RasMol command
<a href="#show"><tt><b>show symmetry</b></tt></a>
display details of the crystal's space group and unit cell axes. The
<tt><b>set unitcell</b></tt>
command is similar the the commands
<a href="#setaxes"><tt><b>set axes</b></tt></a>
and
<a href="#setboundbox"><tt><b>set boundbox</b></tt></a>
that display orthogonal co-ordinate axes and the bounding box
respectively.
<p><hr>
<a name="setvectps"><h3>Set VectPS</h3></a><p>
<pre>
Syntax:  set vectps &lt;boolean&gt;
</pre><p>
The RasMol
<tt><b>vectps</b></tt>
parameter is use to control the way in which the RasMol
<a href="#write"><tt><b>write</b></tt></a>
command generates vector PostScript output files. The command
<tt><b>set vectps on</b></tt>
enables to use of black outlines around spheres and cylinder bonds
producing `cartoon-like' high resolution output. However, the current
implementation of RasMol incorrectly cartoons spheres that are intersected
by more than one other sphere. Hence `ball and stick' models are rendered
correctly by not large spacefilling spheres models. Cartoon outlines
can be disabled, the default, by the command
<tt><b>set vectps off</b></tt>
<p><hr>
<a name="chexprs"><h2>Atom Expressions</h2></a>
RasMol atom expressions uniquely identify an arbitrary group of atoms
within a molecule. Atom expressions are composed of either primitive
expressions,
predefined sets,
comparison operators,
<tt><b>within</b></tt>
expressions,
or logical (boolean) combinations of the above expression types.
<p>
The logical operators allow complex queries to be constructed out of
simpler ones using the standard boolean connectives
<tt><b>and, or</b></tt>
and
<tt><b>not.</b></tt>
These may be abbreviated by the symbols
"&lt;tt&gt;&amp;amp;&lt;/tt&gt;", "&lt;tt&gt;|&lt;/tt&gt;" and "&lt;tt&gt;!&lt;/tt&gt;"
respectively. Parentheses (brackets) may be used to alter the
precedence of the operators. For convenience, a comma may also
be used for boolean disjunction.
<p>
The atom expression is evaluated for each atom, hence
<tt><b>protein and backbone</b></tt>
selects protein bacbone atoms, not the protein and [nucleic] acid
backbone atoms!
<p>
<pre>
Examples:    <b>backbone and not helix</b>
             <b>within( 8.0, ser70 )</b>
             <b>not (hydrogen or hetero)</b>
             <b>not *.FE and hetero</b>
             <b>8, 12, 16, 20-28</b>
             <b>arg, his, lys</b>
</pre>
<ul>
<li><a href="#primitiveexpressions">Primitive Expressions</a>
<li><a href="#predefinedsets">Predefined Sets</a>
<li><a href="#comparisonoperators">Comparison Operators</a>
<li><a href="#withinexpressions"><tt>Within</tt> Expressions</a>
<p><li><a href="#exampleexpressions">Examples</a>
</ul>
<p><hr>
<a name="exampleexpressions"><h3>Example Expressions</h3></a><p>
The following table gives some useful examples of RasMol
atom expressions.
<p>
<pre>
    <b>Expression</b>      <b>Interpretation</b>

    *               All atoms
    cys             Atoms in cysteines
    hoh             Atoms in heterogenous water molecules
    as?             Atoms in either asparagine or aspartic acid
    *120            Atoms at residue 120 of all chains
    *p              Atoms in chain P
    *.n?            Nitrogen atoms
    cys.sg          Sulphur atoms in cysteine residues
    ser70.c?        Carbon atoms in serine-70
    hem*p.fe        Iron atoms in the Heme groups of chain P
</pre>
<p><hr>
<a name="primitiveexpressions"><h3>Primitive Expressions</h3></a><p>
RasMol primitive expressions are the fundamental building blocks
of atom expressions. There are two types of primitive expression.
The first type is used to identify a given residue number or range
of residue numbers. A single residue is identified by its number
(position in the sequence), and a range is specified by lower and
upper bounds separated by a hyphen character. For example
<tt><b>select 5,6,7,8</b></tt>
is also
<tt><b>select 5-8.</b></tt>
Note that this selects the given residue numbers in all macromolecule
chains.
<p>
The second type of primitive expression specifies a sequence of fields
that must match for a given atom. The first part specifies a residue
(or group of residues) and an optional second part specifies the atoms
within those residues. The first part consists of a residue name,
optionally followed by a residue number and/or chain identifier.
<p>
A residue name typically consists of up to three alphabetic characters,
which are case insensitive. Hence the primitive expressions
<tt><b>SER</b></tt>
and
<tt><b>ser</b></tt>
are equivalent, identifying all serine residues.
Residue names that contain non-alphabetic characters, such as
sulphate groups, may be delimited using square brackets, i.e.
<tt><b>[SO4]</b></tt>
<p>
The residue number is the residue's position in the macromolecule
sequence. Negative sequence numbers are permited. For example,
<tt><b>SER70</b></tt>
Care must be taken when specifying both residue name and number,
it the group at the specified position isn't the specified residue
no atoms are selected.
<p>
The chain identifier is typically a single case-insensitive
alphabetic or numeric character. Numeric chain identifiers must
be distinguished or separated from residue numbers by a colon
character. For example,
<tt><b>SER70A</b></tt>
or
<tt><b>SER70:1</b></tt>
<p>
The second part consists of a period character followed by an atom
name.
An atom name may be up to four alphabetic or numeric characters.
<p>
An asterisk may be used as a wild card for a whole field and a
question mark as a single character wildcard.
<p><hr>
<a name="comparisonoperators"><h3>Comparison Operators</h3></a><p>
Parts of a molecule may also be distinguished using equality,
inequality and ordering operators on their properties. The format
of such comparison expression is a property name, followed by a
comparison operator and then an integer value.
<p>
The atom properties that may be used in RasMol are
<tt><b>atomno</b></tt>
for the atom serial number,
<tt><b>elemno</b></tt>
for the atom's atomic number (element),
<tt><b>resno</b></tt>
for the residue number,
<tt><b>radius</b></tt>
for the spacefill radius in RasMol units (or zero if not represented
as a sphere) and
<tt><b>temperature</b></tt>
for the PDB anisotropic temperature value.
<p>
The equality operator is denoted either
"&lt;tt&gt;&lt;b&gt;=&lt;/b&gt;&lt;/tt&gt;" or "&lt;tt&gt;&lt;b&gt;==&lt;/b&gt;&lt;/tt&gt;".
The inequality operator as either
"&lt;tt&gt;&lt;b&gt;&amp;lt;&amp;gt;&lt;/b&gt;&lt;/tt&gt;", "&lt;tt&gt;&lt;b&gt;!=&lt;/b&gt;&lt;/tt&gt;" or
"&lt;tt&gt;&lt;b&gt;/=&lt;/b&gt;&lt;/tt&gt;".
The ordering operators are
"&lt;tt&gt;&lt;b&gt;&amp;lt;&lt;/b&gt;&lt;/tt&gt;"
for less than,
"&lt;tt&gt;&lt;b&gt;&amp;lt;=&lt;/b&gt;&lt;/tt&gt;"
for less than or equal to,
"&lt;tt&gt;&lt;b&gt;&amp;gt;&lt;/b&gt;&lt;/tt&gt;"
for greater than, and
"&lt;tt&gt;&lt;b&gt;&amp;gt;=&lt;/b&gt;&lt;/tt&gt;"
for greater than or equal to.
<p>
<pre>
Examples:    resno < 23
             temperature >= 900
             atomno == 487
</pre>
<p><hr>
<a name="withinexpressions"><h3>Within Expressions</h3></a><p>
A RasMol
<tt><b>within</b></tt>
expression allows atoms to be selected on their proximity to
another set of atoms. A
<tt><b>within</b></tt>
expression takes two parameters separated by a comma and surrounded
by parenthesis. The first argument is an integer value called the
"cut-off" distance of the within expression and the second argument
is any  valid atom expression. The cut-off distance is expressed in
either integer RasMol units or Angstroms containing a decimal point.
An atom is selected if it is within the cut-off distance of any of
the atoms defined by the second argument. This allows complex
expressions to be constructed containing nested
<tt><b>within</b></tt>
expressions.
<p>
For example, the command
<tt><b>select within(3.2,backbone)</b></tt>
selects any atom within a 3.2 Angstrom radius of any atom in a
protein or nucleic acid backbone.
<tt><b>Within</b></tt>
expressions are particularly useful for selecting the atoms
around an active site.
<p><hr>
<a name="predefinedsets"><h3>Predefined Sets</h3></a><p>
RasMol atom expressions may contain predefined sets. These sets
are single keywords that represent portions of a molecule of interest.
Predefined sets are often abbreviations primitive atom expressions,
and in some cases of selecting areas of a molecule that could not
otherwise be distinguished. A list of the currently predefined sets
is given below.
In addition to the sets listed here, RasMol also treats element names
(and their plurals) as predefined sets containing all atoms of that
element type, i.e. the command
<a href="#select"><tt><b>select oxygen</b></tt></a>
is equivalent to the command
<a href="#select"><tt><b>select elemno=8.</b></tt></a>
<p>
<pre>
    <a href="#atset">AT</a>              <a href="#acidicset">Acidic</a>          <a href="#acyclicset">Acyclic</a>
    <a href="#aliphaticset">Aliphatic</a>       <a href="#alphaset">Alpha</a>           <a href="#aminoset">Amino</a>
    <a href="#aromaticset">Aromatic</a>        <a href="#backboneset">Backbone</a>        <a href="#basicset">Basic</a>
    <a href="#bondedset">Bonded</a>          <a href="#buriedset">Buried</a>          <a href="#cgset">CG</a>
    <a href="#chargedset">Charged</a>         <a href="#cyclicset">Cyclic</a>          <a href="#cystineset">Cystine</a>
    <a href="#helixset">Helix</a>           <a href="#heteroset">Hetero</a>          <a href="#hydrogenset">Hydrogen</a>
    <a href="#hydrophobicset">Hydrophobic</a>     <a href="#ionsset">Ions</a>            <a href="#largeset">Large</a>
    <a href="#ligandset">Ligand</a>          <a href="#mediumset">Medium</a>          <a href="#neutralset">Neutral</a>
    <a href="#nucleicset">Nucleic</a>         <a href="#polarset">Polar</a>           <a href="#proteinset">Protein</a>
    <a href="#purineset">Purine</a>          <a href="#pyrimidineset">Pyrimidine</a>      <a href="#selectedset">Selected</a>
    <a href="#sheetset">Sheet</a>           <a href="#sidechainset">Sidechain</a>       <a href="#smallset">Small</a>
    <a href="#solventset">Solvent</a>         <a href="#surfaceset">Surface</a>         <a href="#turnset">Turn</a>
    <a href="#waterset">Water</a>
</pre>
<p><hr>
<a name="atset"><h3>AT Set</h3></a><p>
This set contains the atoms in the complementary nucleotides
adenosine and thymidine (A and T respectively). All nucleotides
are classified as either the set
<tt><b>at</b></tt>
or the set
<a href="#cgset"><tt><b>cg</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>a,t</b></tt>"
and
"<tt><b>nucleic and not cg</b></tt>"
<p><hr>
<a name="acidicset"><h3>Acidic Set</h3></a><p>
The set of acidic amino acids.
These are the residue types Asp and Glu.
All amino acids are classified as either
<tt><b>acidic,</b></tt>
<a href="#basicset"><tt><b>basic</b></tt></a>
<tt><b>or</b></tt>
<a href="#neutralset"><tt><b>neutral.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>asp, glu</b></tt>"
and
"<tt><b>amino and not (basic or neutral)</b></tt>"
<p><hr>
<a name="acyclicset"><h3>Acyclic Set</h3></a><p>
The set of atoms in amino acids not containing a cycle or
ring. All amino acids are classified as either
<a href="#cyclicset"><tt><b>cyclic</b></tt></a>
or
<tt><b>acyclic.</b></tt>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not cyclic</b></tt>"
<p><hr>
<a name="aliphaticset"><h3>Aliphatic Set</h3></a><p>
This set contains the aliphatic amino acids.
These are the amino acids Ala, Gly, Ile, Leu and Val.
This set is equiavlent to the RasMol atom expression
"<tt><b>ala, gly, ile, leu, val</b></tt>"
<p><hr>
<a name="alphaset"><h3>Alpha Set</h3></a><p>
The set of alpha carbons in the protein molecule. This set is
approximately equivalent to the RasMol atom expression
"<tt><b>*.CA</b></tt>"
This command should not be confused with the predefined set
<a href="#helixset"><tt><b>helix</b></tt></a>
which contains the atoms in the amino acids of the protein's
alpha helices.
<p><hr>
<a name="aminoset"><h3>Amino Set</h3></a><p>
This set contains all the atoms contained in amino acid residues.
This is useful for distinguishing the protein from the nucleic
acid and heterogenous atoms in the current molecule database.
<p><hr>
<a name="aromaticset"><h3>Aromatic Set</h3></a><p>
The set of atoms in amino acids containing aromatic rings.
These are the amino acids His, Phe, Trp and Tyr.
Because they contain aromatic rings all members of this
set are member of the predefined set
<a href="#cyclicset"><tt><b>cyclic.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>his, phe, trp, tyr</b></tt>"
and
"<tt><b>cyclic and not pro</b></tt>"
<p><hr>
<a name="backboneset"><h3>Backbone Set</h3></a><p>
This set contains the four atoms of each amino acid that form the
polypeptide N-C-C-O backbone of proteins, and the atoms the sugar
phosphate backbone of nucleic acids.
Use the RasMol predefined sets
<tt><b>protein</b></tt>
and
<tt><b>nucleic</b></tt>
to distinguish between the two forms of backbone.
Atoms in nucleic acids and proteins are either
<tt><b>backbone</b></tt>
or
<a href="#sidechainset"><tt><b>sidechain.</b></tt></a>
This set is equivalent to the RasMol expression
"<tt><b>(protein or nucleic) and not sidechain</b></tt>"
<p>
The predefined set
<tt><b>mainchain</b></tt>
is synonymous with the set
<tt><b>backbone.</b></tt>
<p><hr>
<a name="basicset"><h3>Basic Set</h3></a><p>
The set of basic amino acids.
These are the residue types Arg, His and Lys.
All amino acids are classified as either
<a href="#acidicset"><tt><b>acidic,</b></tt></a>
<tt><b>basic</b></tt>
or
<a href="#neutralset"><tt><b>neutral.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>arg, his, lys</b></tt>"
and
"<tt><b>amino and not (acidic or neutral)</b></tt>"
<p><hr>
<a name="bondedset"><h3>Bonded Set</h3></a><p>
This set contain all the atoms in the current molecule database that
are bonded to atleast one other atom.
<p><hr>
<a name="buriedset"><h3>Buried Set</h3></a><p>
This set contains the atoms in those amino acids that tend
(prefer) to buried inside protein, away from contact with
solvent molecules. This set refers to the amino acids
preference and not the actual solvent acessibility for
the current protein.
All amino acids are classified as either
<a href="#surfaceset"><tt><b>surface</b></tt></a>
or
<tt><b>buried.</b></tt>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not surface</b></tt>"
<p><hr>
<a name="cgset"><h3>CG Set</h3></a><p>
This set contains the atoms in the complementary nucleotides
cytidine and guanoine (C and G respectively). All nucleotides
are classified as either the set
<a href="#atset"><tt><b>at</b></tt></a>
or the set
<tt><b>cg</b></tt>
This set is equivalent to the RasMol atom expressions
"<tt><b>c,g</b></tt>"
and
"<tt><b>nucleic and not at</b></tt>"
<p><hr>
<a name="chargedset"><h3>Charged Set</h3></a><p>
This set contains the charged amino acids. These are the amino
acids that are either
<a href="#acidicset"><tt><b>acidic</b></tt></a>
or
<a href="#basicset"><tt><b>basic.</b></tt></a>
Amino acids are classified as being either
<tt><b>charged</b></tt>
or
<a href="#neutralset"><tt><b>neutral.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>acidic or basic</b></tt>"
and
"<tt><b>amino and not neutral</b></tt>"
<p><hr>
<a name="cyclicset"><h3>Cyclic Set</h3></a><p>
The set of atoms in amino acids containing a cycle or rings.
All amino acids are classified as either
<tt><b>cyclic</b></tt>
or
<a href="#acyclicset"><tt><b>acyclic.</b></tt></a>
This set consists of the amino acids His, Phe, Pro, Trp and Tyr.
The members of the predefined set
<a href="#aromaticset"><tt><b>aromatic</b></tt></a>
are members of this set.
The only cyclic but non-aromatic amino acid is proline.
This set is equivalent to the RasMol atom expressions
"<tt><b>his, phe, pro, trp, tyr</b></tt>"
and
"<tt><b>aromatic or pro</b></tt>"
and
"<tt><b>amino and not acyclic</b></tt>"
<p><hr>
<a name="cystineset"><h3>Cystine Set</h3></a><p>
This set contains the atoms of cysteine residues that form part
of a disulphide bridge, i.e. half cystines. RasMol automatically
determines disulphide bridges, if neither the predefined set
<tt><b>cystine</b></tt>
nor the RasMol
<a href="#ssbonds"><tt><b>ssbonds</b></tt></a>
command have been used since the molecule was loaded. The set of
free cysteines may be determined using the RasMol atom expression
"<tt><b>cys and not cystine</b></tt>"
<p><hr>
<a name="helixset"><h3>Helix Set</h3></a><p>
This set contains all atoms that form part of a protein alpha
helix as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
<a href="#structure"><tt><b>structure</b></tt></a>
command.
<p>
This predefined set should not be confused with the predefined set
<a href="#alphaset"><tt><b>alpha</b></tt></a>
which contains the alpha carbon atoms of a protein.
<p><hr>
<a name="heteroset"><h3>Hetero Set</h3></a><p>
This set contains all the heterogenous atoms in the molecule. These
are the atoms described by HETATM entries in the PDB file. These
typically contain water, cofactors and other solvents and ligands. All
<tt><b>hetero</b></tt>
atoms are classified as either
<a href="#ligandset"><tt><b>ligand</b></tt></a>
or
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms. These heterogenous
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms are further classified as either
<a href="#waterset"><tt><b>water</b></tt></a>
or
<a href="#ionsset"><tt><b>ions.</b></tt></a>
<p><hr>
<a name="hydrogenset"><h3>Hydrogen Set</h3></a><p>
This predefined set contains all the hydrogen and deuterium atoms
of the current molecule. This predefined set is equivalent to the
RasMol atom expression
"<tt><b>elemno=1</b></tt>"
<p><hr>
<a name="hydrophobicset"><h3>Hydrophobic Set</h3></a><p>
This set contains all the hydrophobic amino acids.
These are the amino acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp.
All amino acids are classified as either
<tt><b>hydrophobic</b></tt>
or
<a href="#polarset"><tt><b>polar.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>ala, leu, val, ile, pro, phe, met, trp</b></tt>"
and
"<tt><b>amino and not polar</b></tt>"
<p><hr>
<a name="ionsset"><h3>Ions Set</h3></a><p>
This set contains all the heterogenous phosphate and sulphate ions in
the current molecule data file. A large number of these ions are
sometimes associated with protein and nucleic acid structures determined
by X-ray crystallography. These atoms tend to clutter an image. All
<a href="#heteroset"><tt><b>hetero</b></tt></a>
atoms are classified as either
<a href="#ligandset"><tt><b>ligand</b></tt></a>
or
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms. All
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms are classified as either
<a href="#waterset"><tt><b>water</b></tt></a>
or
<tt><b>ions.</b></tt>
<p><hr>
<a name="largeset"><h3>Large Set</h3></a><p>
All amino acids are classified as either
<a href="#smallset"><tt><b>small,</b></tt></a>
<a href="#mediumset"><tt><b>medium</b></tt></a>
or
<tt><b>large.</b></tt>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not (small or medium)</b></tt>"
<p><hr>
<a name="ligandset"><h3>Ligand Set</h3></a><p>
This set contains all the heterogenous cofactor and ligand moieties that
are contained in the current molecule data file. At this set is defined
to be all
<a href="#heteroset"><tt><b>hetero</b></tt></a>
atoms that are not
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms. Hence this set is equivalent to the RasMol atom expression
"<tt><b>hetero and not solvent</b></tt>"
<p><hr>
<a name="mediumset"><h3>Medium Set</h3></a><p>
All amino acids are classified as either
<a href="#smallset"><tt><b>small,</b></tt></a>
<tt><b>medium</b></tt>
or
<a href="#largeset"><tt><b>large.</b></tt></a>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not (large or small)</b></tt>"
<p><hr>
<a name="neutralset"><h3>Neutral Set</h3></a><p>
The set of neutral amino acids.
All amino acids are classified as either
<a href="#acidicset"><tt><b>acidic,</b></tt></a>
<tt><b>basic</b></tt>
or
<a href="#neutralset"><tt><b>neutral.</b></tt></a>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not (acidic or basic)</b></tt>"
<p><hr>
<a name="nucleicset"><h3>Nucleic Set</h3></a><p>
The set of all atoms in nucleic acids, which consists of the four
nucleotide bases adenosine, cytidine, guanosine and thymidine (A,
C, G and T respectively). All neucleotides are classified as either
<a href="#purineset"><tt><b>purine</b></tt></a>
or
<a href="#pyrimidineset"><tt><b>pyrimidine.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>a,c,g,t</b></tt>"
and
"<tt><b>purine or pyrimidine</b></tt>"
<p><hr>
<a name="polarset"><h3>Polar Set</h3></a><p>
This set contains the polar amino acids.
All amino acids are classified as either
<a href="#hydrophobicset"><tt><b>hydrophobic</b></tt></a>
or
<tt><b>polar.</b></tt>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not hydrophobic</b></tt>"
<p><hr>
<a name="proteinset"><h3>Protein Set</h3></a><p>
The set of all atoms in proteins. This consists of the RasMol
predefined set
<a href="#aminoset"><tt><b>amino</b></tt></a>
and common post-translation modifications.
<p><hr>
<a name="purineset"><h3>Purine Set</h3></a><p>
The set of purine nucleotides.
These are the bases adenosine and guanosine (A and G respectively).
All nucleotides are either
<tt><b>purines</b></tt>
or
<a href="#pyrimidineset"><tt><b>pyrimidines.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>a,g</b></tt>"
and
"<tt><b>nucleic and not purine</b></tt>"
<p><hr>
<a name="pyrimidineset"><h3>Pyrimidine Set</h3></a><p>
The set of pyrimidine nucleotides.
These are the bases cytidine and thymidine (C and T respectively).
All nucleotides are either
<a href="#purineset"><tt><b>purines</b></tt></a>
or
<tt><b>pyrimidines.</b></tt>
This set is equivalent to the RasMol atom expressions
"<tt><b>c,t</b></tt>"
and
"<tt><b>nucleic and not pyrimidine</b></tt>"
<p><hr>
<a name="selectedset"><h3>Selected Set</h3></a><p>
This set contains the set of atoms in the currently selected
region. The currently selected region is defined by the preceding
<a href="#select"><tt><b>select</b></tt></a>
or
<a href="#restrict"><tt><b>restrict</b></tt></a>
command and not the atom expression containing the
<tt><b>selected</b></tt>
keyword.
<p><hr>
<a name="sheetset"><h3>Sheet Set</h3></a><p>
This set contains all atoms that form part of a protein beta
sheet as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
<a href="#structure"><tt><b>structure</b></tt></a>
command.
<p><hr>
<a name="sidechainset"><h3>Sidechain Set</h3></a><p>
This set contains the functional sidechains of any amino acids
and the base of each nucleotide. These are the atoms not part of
the polypeptide N-C-C-O backbone of proteins or the sugar
phosphate backbone of nucleic acids.
Use the RasMol predefined sets
<tt><b>protein</b></tt>
and
<tt><b>nucleic</b></tt>
to distinguish between the two forms of sidechain.
Atoms in nucleic acids and proteins are either
<a href="#backboneset"><tt><b>backbone</b></tt></a>
or
<tt><b>sidechain.</b></tt>
This set is equivalent to the RasMol expression
"<tt><b>(protein or nucleic) and not backbone</b></tt>"
<p><hr>
<a name="smallset"><h3>Small Set</h3></a><p>
All amino acids are classified as either
<tt><b>small,</b></tt>
<a href="#mediumset"><tt><b>medium</b></tt></a>
or
<a href="#largeset"><tt><b>large.</b></tt></a>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not (medium or large)</b></tt>"
<p><hr>
<a name="solventset"><h3>Solvent Set</h3></a><p>
This set contains the solvent atoms in the molecule co-ordinate file.
These are the heterogenous water molecules, phosphate and sulphate
ions. All
<a href="#heteroset"><tt><b>hetero</b></tt></a>
atoms are classified as either
<a href="#ligandset"><tt><b>ligand</b></tt></a>
or
<tt><b>solvent</b></tt>
atoms. All
<tt><b>solvent</b></tt>
atoms are classified as either
<a href="#waterset"><tt><b>water</b></tt></a>
or
<a href="#ionsset"><tt><b>ions.</b></tt></a>
This set is equivalent to the RasMol atom expressions
"<tt><b>hetero and not ligand</b></tt>"
and
"<tt><b>water or ions</b></tt>"
<p><hr>
<a name="surfaceset"><h3>Surface Set</h3></a><p>
This set contains the atoms in those amino acids that tend
(prefer) to be on the surface of proteins, in contact with
solvent molecules. This set refers to the amino acids
preference and not the actual solvent accessibility for
the current protein.
All amino acids are classified as either
<tt><b>surface</b></tt>
or
<a href="#buriedset"><tt><b>buried.</b></tt></a>
This set is equivalent to the RasMol atom expression
"<tt><b>amino and not buried</b></tt>"
<p><hr>
<a name="turnset"><h3>Turn Set</h3></a><p>
This set contains all atoms that form part of a protein turns
as determined by either the PDB file author or Kabsch and
Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists.
Otherwise, it uses the DSSP algorithm as used by the RasMol
<a href="#structure"><tt><b>structure</b></tt></a>
command.
<p><hr>
<a name="waterset"><h3>Water Set</h3></a><p>
This set contains all the heterogenous water molecules in the current
database. A large number of water molecules are sometimes associated
with protein and nucleic acid structures determined by X-ray
crystallography. These atoms tend to clutter an image.
All
<a href="#heteroset"><tt><b>hetero</b></tt></a>
atoms are classified as either
<a href="#ligandset"><tt><b>ligand</b></tt></a>
or
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms. The
<a href="#solventset"><tt><b>solvent</b></tt></a>
atoms are further classified as either
<tt><b>water</b></tt>
or
<a href="#ionsset"><tt><b>ions.</b></tt></a>
<p><hr>
<a name="chcolours"><h2>Colour Schemes</h2></a>
The RasMol
<a href="#colour"><tt><b>colour</b></tt></a>
command allows different objects (such as atoms, bonds and ribbon segments)
to be given a specified colour. Typically this colour is either a RasMol
predefined colour name or an RGB triple. Additionally RasMol also supports
<a href="#cpkcolours"><tt><b>cpk,</b></tt></a>
<a href="#aminocolours"><tt><b>amino,</b></tt></a>
<a href="#chaincolours"><tt><b>chain,</b></tt></a>
<a href="#groupcolours"><tt><b>group,</b></tt></a>
<a href="#shapelycolours"><tt><b>shapely,</b></tt></a>
<a href="#structurecolours"><tt><b>structure,</b></tt></a>
<a href="#temperaturecolours"><tt><b>temperature,</b></tt></a>
<a href="#chargecolours"><tt><b>charge</b></tt></a>
and
<a href="#usercolours"><tt><b>user</b></tt></a>
colour schemes for atoms, a
<a href="#hbondtypecolours"><tt><b>hbond type</b></tt></a>
colour scheme for hydrogen bonds and
<a href="#potentialcolours"><tt><b>electrostatic potential</b></tt></a>
colour scheme for dot surfaces.
The currently predefined colour
names are
listed below with their corresponding RGB triplet.
<p>
<pre>
    blue         [0,0,255]          black        [0,0,0]
    cyan         [0,255,255]        green        [0,255,0]
    greenblue    [46,139,87]        magenta      [255,0,255]
    orange       [255,165,0]        purple       [160,32,240]
    red          [255,0,0]          redorange    [255,69,0]
    violet       [238,130,238]      white        [255,255,255]
    yellow       [255,255,0]
</pre>
<p><hr>
<a name="aminocolours"><h3>Amino Colours</h3></a><p>
The RasMol
<tt><b>amino</b></tt>
colour scheme colours amino acids according to traditional amino acid
properties. The purpose of colouring is to identify amino acids in an
unusual or surprising environment. The outer parts of a protein that are
polar are visible (bright) colours and non-polar residues darker. Most
colours are hallowed by tradition. This colour scheme is similar to the
<a href="#shapelycolours"><tt><b>shapely</b></tt></a>
scheme.
<p>
<pre>
   ASP,GLU bright red [230,10,10]     CYS,MET     yellow [230,230,0]
   LYS,ARG blue       [20,90,255]     SER,THR     orange [250,150,0]
   PHE,TYR mid blue   [50,50,170]     ASN,GLN     cyan   [0,220,220]
   GLY     light grey [235,235,235]   LEU,VAL,ILE green  [15,130,15]
   ALA     dark grey  [200,200,200]   TRP         pink   [180,90,180]
   HIS     pale blue  [130,130,210]   PRO         flesh  [220,150,130]
</pre>
<p><hr>
<a name="chaincolours"><h3>Chain Colours</h3></a><p>
The RasMol
<tt><b>chain</b></tt>
colour scheme assigns each macromolecular chain a unique colour. This
colour scheme is particularly useful for distinguishing the parts of
multimeric structure or the individual `strands' of a DNA chain.
<p><hr>
<a name="cpkcolours"><h3>CPK Colours</h3></a><p>
The RasMol
<tt><b>cpk</b></tt>
colour scheme is based upon the colours of the popular plastic
spacefilling models which were developed by Corey, Pauling and later
improved by Kultun. This colour scheme colour `atom' objects by the
atom (element) type. This is the scheme conventionally used by chemists.
The assignment of element type to colours is given below.
<p>
<pre>
    Carbon       light grey       Chlorine         green
    Oxygen       red              Bromine, Zinc    brown
    Hydogen      white            Sodium           blue
    Nitrogen     light blue       Iron             purple
    Sulphur      yellow           Calcium, Metals  dark grey
    Phosphorous  orange           Unknown          deep pink
</pre>
<p><hr>
<a name="groupcolours"><h3>Group Colours</h3></a><p>
The RasMol
<tt><b>group</b></tt>
colour scheme colour codes residues by their position in a macromolecular
chain. Each chain is drawn as a smooth spectrum from blue through green,
yellow and orange to red. Hence the N terminus of proteins and 5' terminus
of nucleic acids are coloured red and the C terminus of proteins and 3'
terminus of nucleic acids are drawn in blue. If a chain has a large number
of heterogenous molecules associated with it, the macromolecule may not be
drawn in the full `range' of the spectrum.
<p><hr>
<a name="shapelycolours"><h3>Shapely Colours</h3></a><p>
The RasMol
<tt><b>shapely</b></tt>
colour scheme colour codes residues by amino acid property. This scheme
is based upon Bob Fletterick's "Shapely Models". Each amino acid and
nucleic acid residue is given a unique colour. The
<tt><b>shapely</b></tt>
colour scheme is used by David Bacon's Raster3D program. This colour
scheme is similar to the
<a href="#aminocolours"><tt><b>amino</b></tt></a>
colour scheme.
<p><hr>
<a name="structurecolours"><h3>Structure Colours</h3></a><p>
The RasMol
<tt><b>structure</b></tt>
colour scheme colours the molecule by protein secondary structure.
Alpha helices are coloured magenta, [240,0,128], beta sheets are
coloured yellow, [255,255,0], turns are coloured pale blue, [96,128,255]
and all other residues are coloured white. The secondary structure
is either read from the PDB file (HELIX and SHEET records), if available,
or determined using Kabsch and Sander's DSSP algorithm. The RasMol
<a href="#structure"><tt><b>structure</b></tt></a>
command may be used to force DSSP's structure assignment to be used.
<p><hr>
<a name="temperaturecolours"><h3>Temperature Colours</h3></a><p>
The RasMol
<tt><b>temperature</b></tt>
colour scheme colour codes each atom according to the anisotropic
temperature (beta) value stored in the PDB file. Typically this gives
a measure of the mobility/uncertainty of a given atom's position. High
values are coloured in warmer (red) colours and lower values in colder
(blue) colours. This feature is often used to associate a "scale" value
[such as amino acid variability in viral mutants] with each atom in a
PDB file, and colour the molecule appropriately.
<p>
The difference between the
<tt><b>temperature</b></tt>
and
<a href="#chargecolours"><tt><b>charge</b></tt></a>
colour schemes is that increasing temperature values proceed from blue
to red, whereas increasing charge valuse go from red to blue.
<p><hr>
<a name="chargecolours"><h3>Charge Colours</h3></a><p>
The RasMol
<tt><b>charge</b></tt>
colour scheme colour codes each atom according to the charge value
stored in the input file (or beta factor field of PDB files). High
values are coloured in blue (positive) and lower values coloured in
red (negative). Rather than use a fixed scale this scheme determines
the maximum and minimum values of the charge/temperature field and
interpolates from red to blue appropriately. Hence, green cannot be
assumed to be `no net charge' charge.
<p>
The difference between the
<a href="#chargecolours"><tt><b>charge</b></tt></a>
and
<tt><b>temperature</b></tt>
colour schemes is that increasing temperature values proceed from blue
to red, whereas increasing charge valuse go from red to blue.
<p>
If the charge/temperature field stores reasonable values it is possible
to use the RasMol
<a href="#potentialcolours"><tt><b>colour dots potential</b></tt></a>
command to colour code a dot surface (generated by the
<a href="#dots"><tt><b>dots</b></tt></a>
command) by electrostatic potential.
<p><hr>
<a name="usercolours"><h3>User Colours</h3></a><p>
The RasMol
<tt><b>user</b></tt>
colour scheme allows RasMol to use the colour scheme stored in the
PDB file. The colours for each atom are stored in COLO records placed
in the PDB data file. This convention was introduced by David Bacon's
Raster3D program.
<p><hr>
<a name="hbondtypecolours"><h3>HBond Type Colours</h3></a><p>
The RasMol
<tt><b>type</b></tt>
colour scheme applies only to hydrogen bonds, hence is used in the command
"<tt><b>colour hbonds type</b></tt>"
This scheme colour codes each hydrogen bond according to the
distance along a protein chain between hydrogen bond donor and acceptor.
This schematic representation was introduced by Belhadj-Mostefa and
Milner-White. This representation gives a good insight into protein
secondary structure (hbonds forming alpha helices appear red, those
forming sheets appear yellow and those forming turns appear magenta).
<p>
<pre>
      Offset    Colour    Triple
        +2      white     [255,255,255]
        +3      magenta   [255,0,255]
        +4      red       [255,0,0]
        +5      orange    [255,165,0]
        -3      cyan      [0,255,255]
        -4      green     [0,255,0]
      default   yellow    [255,255,0]
</pre>
<p><hr>
<a name="potentialcolours"><h3>Potential Colours</h3></a><p>
The RasMol
<tt><b>potential</b></tt>
colour scheme applies only to dot surfaces, hence is used in the command
"<tt><b>colour dots potential</b></tt>"
This scheme colours each currently displayed dot by the electrostatic
potential at that point in space. This potential is calculated using
Coulomb's law taking the temperature/charge field of the input file to
be the charge assocated with that atom. This is the same interpretation
used by the
<a href="#chargecolours"><tt><b>colour charge</b></tt></a>
command. Like the
<a href="#chargecolours"><tt><b>charge</b></tt></a>
colour scheme low values are blue/white and high values are red.
The table below shows the static assignment of colours using a
dielectric constant value of 10.
<p>
<pre>
     25 < V          red       [255,0,0]
     10 < V <  25    orange    [255,165,0]
      3 < V <  10    yellow    [255,255,0]
      0 < V <   3    green     [0,255,0]
     -3 < V <   0    cyan      [0,255,255]
    -10 < V <   3    blue      [0,0,255]
    -25 < V < -10    purple    [160,32,240]
          V < -25    white     [255,255,255]
</pre>
<p>
